sdk/eventhub/azure-eventhubs/samples/sync_samples/recv_with_partition_manager.py

```diff
@@ -82,7 +51,16 @@ def receive_forever(client):
         retry_total=RETRY_TOTAL  # num of retry times if receiving from EventHub has an error.
     )
 
-    with consumer_client:
-        receiving_time = 60
-        receive_for_a_while(consumer_client, receiving_time)
-        # receive_forever(consumer_client)
+    try:
+        with consumer_client:
+            """
+            Without specified partition_id, the receive will try to receive events from all partitions and if provided with
+            partition manager, the client will load-balance partition assignment with other EventHubConsumerClient instances
+            which also try to receive events from all partitions and use the same storage resource.
+            """
+            consumer_client.receive(event_handler=event_handler, consumer_group='$Default')
+            # With specified partition_id, load-balance will be disabled
+            # client.receive(event_handler=event_handler, consumer_group='$Default', partition_id='0')
+    except KeyboardInterrupt:
+        print('Stop receiving.')
+        consumer_client.close()
```

Comment:
Is the only way to stop receiving with an interrupt?

Reply:
user can create a thread executing the `.receive` method, and then call `.close` to stop the receiving.